### ISSUE TITLE
JRE11 - JVM crash consistently with traceassert

### ISSUE THREAD

**Issue by yaakov-berkovitch** [eclipse-openj9/openj9#12917]
Java -version output
--------------------

```
openjdk version "11.0.11" 2021-04-20
OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (build 11.0.11+9)
Eclipse OpenJ9 VM AdoptOpenJDK-11.0.11+9 (build openj9-0.26.0, JRE 11 Linux amd64-64-Bit Compressed References 20210421_975 (JIT enabled, AOT enabled)
OpenJ9   - b4cc246d9
OMR      - 162e6f729
JCL      - 7796c80419 based on jdk-11.0.11+9)

```

Summary of problem
------------------

The application failed during launching, and the following is dumped to the stderr:
```
*** Invalid JIT return address 0000000000000000 in 0000000000030C10

08:52:38.707 0x30900    j9vm.249    *   ** ASSERTION FAILED ** at ../../../../../openj9/runtime/vm/swalk.c:1565: ((0 ))
JVMDUMP039I Processing dump event "traceassert", detail "" at 2021/06/10 11:52:38 - please wait.
JVMDUMP032I JVM requested System dump using '/data/ms-metro/logs//core.20210610.115238.19481.0001.dmp' in response to an event
JVMDUMP010I System dump written to /data/ms-metro/logs//core.20210610.115238.19481.0001.dmp
JVMDUMP032I JVM requested Java dump using '/data/ms-metro/logs//javacore.20210610.115238.19481.0002.txt' in response to an event
JVMDUMP010I Java dump written to /data/ms-metro/logs//javacore.20210610.115238.19481.0002.txt
JVMDUMP032I JVM requested Snap dump using '/data/ms-metro/logs//Snap.20210610.115238.19481.0003.trc' in response to an event
JVMDUMP010I Snap dump written to /data/ms-metro/logs//Snap.20210610.115238.19481.0003.trc
JVMDUMP013I Processed dump event "traceassert", detail "".
```
**Update:**
The application is configured with Xmx=32GB. I tried to run it with 20GB and it run.

Diagnostic files
----------------

I attached the diagnostics files. The dmp file size is 170MB after zip, so let me know if you need it also.

[jvm11-crash-withtraceassert.zip](https://github.com/eclipse-openj9/openj9/files/6629897/jvm11-crash-withtraceassert.zip)

**Comment by pshipton** [comment]
I wonder if this has been fixed already. Can you please give the following a try and see if it sill occurs. https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Nightly/16/OpenJ9-JDK11-x86-64_linux-20210605-011624.tar.gz

The fact that it fails with 32GB and passes with 20GB makes me think it could be related to https://github.com/eclipse-openj9/openj9/pull/12905 which isn't merged yet, but there is a build containing the current state of the fix here which you can try https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Build_JDK11_x86-64_linux_Personal/57/OpenJ9-JDK11-x86-64_linux-20210610-105743.tar.gz

**Comment by yaakov-berkovitch** [comment]
@pshipton Will try and let you know.

**Comment by yaakov-berkovitch** [comment]
I installed the build and it indeed fixed the issue. Really impressed by the fact that the issue you mentioned and ours occurred almost in the same time.
Do you have any schedule when it will be merge, and also when it will be part of official version ?

**Comment by pshipton** [comment]
I assume you only tried the second build I suggested? It would be worth also trying the first build as the problem may not be related to https://github.com/eclipse-openj9/openj9/pull/12905 and it would be good to know.

The first build is equivalent to a Milestone 1 for the next release 0.27, so if that works the fix is already included in the release.

I expect https://github.com/eclipse-openj9/openj9/pull/12905 to be merged today or tomorrow, and to be added afterwards to the 0.27 release content.

The release activities commence July 20 once OpenJDK finalizes a release build, and should be complete by July 23 but there may be delays for various reasons. One reason is because https://adoptopenjdk.net/ is no longer creating builds. It's been replace by https://adoptium.net/ which doesn't support building OpenJ9. We are working to have another release build channel in place by July 23.

**Comment by yaakov-berkovitch** [comment]
Correct I tried the second build. I will try the first one also and update.

**Comment by yaakov-berkovitch** [comment]
The first build caused a crash:
```
Unhandled exception
Type=Segmentation error vmState=0x00000000
J9Generic_Signal_Number=00000018 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000001
Handler1=00007F8D491C8C70 Handler2=00007F8D48F279F0 InaccessibleAddress=000000002F677218
RDI=0000000000000000 RSI=0000000001A120D0 RAX=000000002F677200 RBX=000000002F677200
RCX=00000000001E0001 RDX=0000000001950500 R8=0000000001950500 R9=00000000021E0001
R10=0000000000DEF400 R11=0000000FFE9A0338 R12=0000000000F5D3B0 R13=0000000000F5D3B8
R14=00007F8D468C8AD9 R15=00007F8D4AC003A0
RIP=00007F8D4922C4E1 GS=0000 FS=0000 RSP=00007F8D4AC00020
EFlags=0000000000010246 CS=0033 RBP=0000000001A137E8 ERR=0000000000000004
TRAPNO=000000000000000E OLDMASK=0000000000000000 CR2=000000002F677218
xmm0 0000000000000005 (f: 5.000000, d: 2.470328e-323)
xmm1 00000008241a43f0 (f: 605701120.000000, d: 1.727522e-313)
xmm2 00007f8d4ac003d0 (f: 1254097920.000000, d: 6.929015e-310)
xmm3 00007f8d4ac003d0 (f: 1254097920.000000, d: 6.929015e-310)
xmm4 000000003ca1b654 (f: 1017230912.000000, d: 5.025789e-315)
xmm5 000000003963f3ba (f: 962851776.000000, d: 4.757120e-315)
xmm6 000000003d23e788 (f: 1025763200.000000, d: 5.067944e-315)
xmm7 00000000016db3a8 (f: 23966632.000000, d: 1.184109e-316)
xmm8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm9 00000000016db348 (f: 23966536.000000, d: 1.184104e-316)
xmm10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
xmm11 ca62c1d6ca62c1d6 (f: 3395469824.000000, d: -2.193092e+50)
xmm12 0000000000000001 (f: 1.000000, d: 4.940656e-324)
xmm13 08090a0b0c0d0e0f (f: 202182160.000000, d: 5.924543e-270)
xmm14 6800000000000000 (f: 0.000000, d: 9.124881e+192)
xmm15 1cf41107a36ace6a (f: 2741685760.000000, d: 3.323185e-169)
Module=/usr/lib/jvm/adoptopenjdk-11-openj9-nightly-5-6-21/jdk/lib/default/libj9vm29.so
Module_base_address=00007F8D4918A000
Target=2_90_20210605_16 (Linux 3.10.0-1160.6.1.el7.x86_64)
CPU=amd64 (32 logical CPUs) (0x1f701cc000 RAM)
----------- Stack Backtrace -----------
(0x00007F8D4922C4E1 [libj9vm29.so+0xa24e1])
(0x00007F8D49219D95 [libj9vm29.so+0x8fd95])
(0x00007F8D492C5152 [libj9vm29.so+0x13b152])
---------------------------------------
JVMDUMP039I Processing dump event "gpf", detail "" at 2021/06/10 18:37:16 - please wait.
JVMDUMP032I JVM requested System dump using '/data/ms-metro/logs//core.20210610.183716.14702.0001.dmp' in response to an event
JVMDUMP010I System dump written to /data/ms-metro/logs//core.20210610.183716.14702.0001.dmp
JVMDUMP032I JVM requested Java dump using '/data/ms-metro/logs//javacore.20210610.183716.14702.0002.txt' in response to an event
JVMDUMP010I Java dump written to /data/ms-metro/logs//javacore.20210610.183716.14702.0002.txt
JVMDUMP032I JVM requested Snap dump using '/data/ms-metro/logs//Snap.20210610.183716.14702.0003.trc' in response to an event
JVMDUMP010I Snap dump written to /data/ms-metro/logs//Snap.20210610.183716.14702.0003.trc
JVMDUMP032I JVM requested JIT dump using '/data/ms-metro/logs//jitdump.20210610.183716.14702.0004.dmp' in response to an event
JVMDUMP051I JIT dump occurred in 'main' thread 0x0000000000030900
JVMDUMP013I Processed dump event "gpf", detail "".

```

**Comment by pshipton** [comment]
Thanks for checking.

**Comment by pshipton** [comment]
Anything below 27GB should work as 27GB is the point where the automatic switch to 4 bit shift occurs, and it's the 4 bit shift which isn't working.

**Comment by pshipton** [comment]
The fix is added for the 0.27 release https://github.com/eclipse-openj9/openj9/pull/12924
